src/vs/workbench/contrib/chat/browser/chatEditing/chatEditingModifiedNotebookEntry.ts

```diff
@@ -492,11 +493,30 @@ export class ChatEditingModifiedNotebookEntry extends AbstractChatEditingModifie
 			confirmBeforeUndo: false,
 			undo: async () => {
 				last = createSnapshot(this.modifiedModel, transientOptions, outputSizeLimit);
-				restoreSnapshot(this.modifiedModel, initial);
+				this._isEditFromUs = true;
+				try {
+					restoreSnapshot(this.modifiedModel, initial);
+					restoreSnapshot(this.originalModel, initial);
```

Comment:
Shouldn't we restore the original model to the original state, i.e. we might want to capture the original state of the original model in line 484 as follows
```typescript
let originalInitial = createSnapshot(this.originalModel, transientOptions, outputSizeLimit);
```

& then 
```tyepscript
restoreSnapshot(this.originalModel, originalInitial)
```

Similarly, shouldn't we capture the current state of original in line 495 as follows
```typescript
originalLast = createSnapshot(this.originalModel, transientOptions, ....)
```

& then in line 512, we change it to `restoreSnapshot(this.originalModel, originalLast)`

TLDR: I do not think we should restore the original & modifed back to the exact same state, else in doing so we end up with 0 differences between the two.

Reply:
just verified that it mostly acts this way - undo and redo the change, but don't put it back in a modified state with changes to approve